### ISSUE TITLE
Computing in-degree histogram of training minibatch

### ISSUE THREAD

**Issue by hash-ir** [lukecavabarrett/pna#11]
Hi, 

I have a single graph dataset that is divided into train, validation, and test sets using PyTorch geometric's `RandomNodeSplit`. This basically adds train, validation, and test masks while keeping the edge index the same. The data object looks like this:
```python
Data(x=[10000, 100], edge_index=[2, 200000], edge_attr=[200000, 20], y=[10000], train_mask=[10000], val_mask=[10000], test_mask=[10000])
```
Since the edge index is for the whole graph, how do I compute the in-degree histogram for normalizing the degree scalers? I am using PyTorch geometric's `degree` function to do this. In the examples, I usually see the molecular datasets with multiple graphs each having its own edge index.

**Comment by hash-ir** [comment]
I solved the problem by using the train, validation, and test subgraphs that hold the `edge_index` corresponding to the nodes in these three sets. Here is how I did it in case anyone has the same problem:
```python
import torch
from torch_geometric.utils import degree

tr_subgraph = data.subgraph(data.train_mask)
# compute the in-degree of all the training nodes
d = degree(index=tr_subgraph.edge_index[1], num_nodes=tr_subgraph.num_nodes, dtype=torch.long)

# get the maximum in-degree, this will be the size of the histogram tensor
max_degree = d.max().item()

# create the in-degree histogram tensor
deg = torch.zeros(max_degree + 1, dtype=torch.long)
deg += torch.bincount(d, minlength=deg.numel())  
```
The last step can be modified for multiple graphs in the dataset by iterating over each and summing the bincounts.
